test/test_hdf5.cpp

```diff
@@ -0,0 +1,140 @@
+//
+// Created by Izzet Yildirim <iyildirim@hawk.iit.edu> on 9/4/24.
+//
+#include <brahma/brahma.h>
+#ifdef BRAHMA_ENABLE_HDF5
+#include <hdf5.h>
+
+namespace brahma {
+class POSIXTest : public POSIX {
```

Comment:
If it is hdf5 specific do we need others?

Reply:
It was printing some warning about POSIX being used but not captured (via GOTCHA?)

Reply:
This is the part about conditional binding I was talking about. It's binding everything all the time. It should just be a warning and not an error. Can you find the source? I suggest we don't go through a warning; we can switch to debug.